x-pack/legacy/plugins/reporting/server/legacy.ts

```diff
@@ -4,35 +4,118 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import { Legacy } from 'kibana';
-import { PluginInitializerContext } from 'src/core/server';
+import crypto from 'crypto';
+import { get } from 'lodash';
+import { CoreSetup, PluginInitializerContext } from 'src/core/server';
 import { SecurityPluginSetup } from '../../../../plugins/security/server';
 import { ReportingPluginSpecOptions } from '../types';
 import { plugin } from './index';
-import { LegacySetup, ReportingStartDeps } from './types';
+import { LegacySetup, ReportingConfig, ReportingConfigType, ReportingStartDeps } from './types';
 
 const buildLegacyDependencies = (
   server: Legacy.Server,
   reportingPlugin: ReportingPluginSpecOptions
-): LegacySetup => ({
-  config: server.config,
-  info: server.info,
-  route: server.route.bind(server),
-  plugins: {
-    elasticsearch: server.plugins.elasticsearch,
-    xpack_main: server.plugins.xpack_main,
-    reporting: reportingPlugin,
-  },
-});
+): LegacySetup => {
+  return {
+    route: server.route.bind(server),
+    config: server.config,
+    plugins: {
+      xpack_main: server.plugins.xpack_main,
+      reporting: reportingPlugin,
+    },
+  };
+};
+
+const addConfigDefaults = (core: CoreSetup, baseConfig: ReportingConfigType) => {
+  // encryption key
+  let encryptionKey = baseConfig.encryptionKey;
+  if (encryptionKey === undefined) {
+    // FIXME log warning
```

Comment:
we can probably just use server.log here. The future of this function is to be moved to the NP as part of a config module that maps the config observable: https://github.com/elastic/kibana/pull/61873/files#diff-f0e5da6abfcea30361b4fb2c632ecc70R15